tests/cpp/test_resize.cpp

```diff
@@ -3677,4 +3677,588 @@ TEST_F(ResizeTest, Chunk_Uneven) {
   EXPECT_EQ(out_tensors.back().numel(), 1);
 }
 
+// Schedule a slice with the loop domain derived from the producer
+// domain. See PR #2897.
+// Note that the IdModel-based indexing is automatically enabled as
+// there are tensors that have non-trivial loop domains as defined by
+// requiresIdModel in lower2device.cpp.
+TEST_F(ResizeTest, SliceScheduledLikeProducer) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  std::vector<int64_t> shape({100});
+
+  // concrete shapes to avoid dynamic Fusion
+  auto tv0 = makeConcreteTensor(shape);
+  fusion.addInput(tv0);
+
+  auto tv1 =
+      slice(tv0, {{IrBuilder::create<Val>(1L), IrBuilder::create<Val>(99)}});
+
+  auto tv2 = set(tv1);
+
+  fusion.addOutput(tv2);
+
+  tv1->setLoopDomain(tv1->getRootDomain());
+
+  auto tv2_loop_id = tv0->getLoopDomain().at(0)->cloneWithoutRFactor();
+
+  IrBuilder::create<Resize>(
+      tv2->getLogicalDomain().at(0),
+      tv2_loop_id,
+      IrBuilder::create<Val>(-1, DataType::Index),
+      IrBuilder::create<Val>(-1, DataType::Index));
+
+  tv2->setLoopDomain({tv2_loop_id});
```

Comment:
naive question: why are we manually  adding the `Resize` with (-1, -1) here? Wasn't `tv2->getLogicalDomain().at(0)` already equivalent to `tv1->getLogicalDomain().at(0)` and hence can't we just re-use the loop domain as `tv2->setLoopDomain(tv1->getLogicalDomain())`?